websocket/src/websocket.cpp

```diff
@@ -264,13 +269,20 @@ static int LuaConnect(lua_State* L)
         return DM_LUA_ERROR("The web socket module isn't initialized");
 
     const char* url = luaL_checkstring(L, 1);
+    lua_Number timeout = dmScript::CheckTableNumber(L, 2, "timeout", 3000);
+    const char* custom_headers = dmScript::CheckTableString(L, 2, "headers", 0);
 
-    // long playedTime = luaL_checktable_number(L, 2, "playedTime", -1);
-    // long progressValue = luaL_checktable_number(L, 2, "progressValue", -1);
-    // char *description = luaL_checktable_string(L, 2, "description", NULL);
-    // char *coverImage = luaL_checktable_string(L, 2, "coverImage", NULL);
+    if (custom_headers != 0)
+    {
+        if (strstr(custom_headers, "\r\n\r\n") != 0)
```

Comment:
Adding the check here to make sure it won't mess up the handshake writing later.